### PR TITLE
fix(test): added unit test for secret manager

### PR DESCRIPTION
**Pull request by madhavilosetty-intel**
## PR Checklist

- [X] Unit Tests have been added for new changes
- [ ] API tests have been updated if applicable
- [ ] All commented code has been removed
- [ ] If you've added a dependency, you've ensured license is compatible with Apache 2.0 and clearly outlined the added dependency.

### CODE REVIEW

File: src/test/secretManagerService.test.ts [hunk 1]
```diff
@@ -0,0 +1,242 @@
+import { SecretManagerService } from '../utils/SecretManagerService'
+import { logger as log } from '../utils/logger'
+import { configType } from '../models/Config'
+import NodeVault from 'node-vault'
+// Parsing configuration
+const config: configType = {
```
Comment (rsdmike) [inline src/test/secretManagerService.test.ts:6]: i think we need to consolidate this config to a single place, import it into each test file and change we need. I don't think we need to copy/paste this into every test file like we're currently doing.

File: src/test/secretManagerService.test.ts [hunk 2]
```diff
@@ -0,0 +1,242 @@
...
+    web_tls_cert_key: '',
+    web_tls_cert_ca: ''
+}
+
+afterEach(() => {
+  jest.clearAllMocks()
+})
+
+test('should get a secret for specific given key of a path', async () => {
+    const secretManagerService: SecretManagerService = new SecretManagerService(config, log)
```
Comment (rsdmike) [inline src/test/secretManagerService.test.ts:72]: this should be able to be moved to a beforeEach yes?

File: src/utils/SecretManagerService.ts [hunk 3]
```diff
@@ -56,33 +58,35 @@ export class SecretManagerService implements ISecretManagerService {
...
   async getAMTCredentials (path: string): Promise<string[]> {
-    try {
-      const user = 'admin'
-      const secret: any = await this.getSecretAtPath(`devices/${path}`)
-      const amtpass = secret.data.AMT_PASSWORD
-      return [user, amtpass]
-    } catch (error) {
-      this.logger.error('Error while retrieving device credentials :', error)
+    const user = 'admin'
+    const secret: any = await this.getSecretAtPath(`devices/${path}`)
```
Comment (rsdmike) [inline src/utils/SecretManagerService.ts:75]: maybe we should type this?

**Review by rsdmike — CHANGES_REQUESTED** [review]

File: src/utils/SecretManagerService.ts (hunk 3 above)
Comment (madhavilosetty-intel) [reply src/utils/SecretManagerService.ts]: done

File: src/test/secretManagerService.test.ts (hunk 2 above)
Comment (madhavilosetty-intel) [reply src/test/secretManagerService.test.ts:21]: done

File: src/test/secretManagerService.test.ts (hunk 1 above)
Comment (madhavilosetty-intel) [reply src/test/secretManagerService.test.ts]: done

**Review by rsdmike — APPROVED** [review]

**Pull request merged**